### PR TITLE
Feature/luisaparicio/search components

### PR DESCRIPTION
**Pull request by ghost**
# Description

Creación de component para listar los resultados de búsqueda y cambios de comportamiento en el search bar para que aparezca y desaparezca al momento de presionar botón.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New component (non-breaking change which adds component)
- [ ] New hook (non-breaking change which adds hook)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Maintenance (repo config)
- [ ] Documentation

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [X] Storybook
- [ ] Unit testing

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] The target of this commit is 'dev'
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/searchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+export const SearchBar = ({
+  clearSearch,
+  ...props
+}: SearchBarProps & TextFieldProps) => {
+  const classes = useStyles();
+  const [showIcon, setShowIcon] = useState<boolean>(false);
+  const [showInputField, setshowInputField] = useState<boolean>(false);
+
+  const handleInputChange = () => {
```
Comment (gsulloa) [inline src/components/searchBar.tsx]: agregar useCallback

File: src/components/searchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          className={showInputField ? classes.searchButtonHide : ''}
+        >
+          <SearchIcon />
+        </IconButton>
+      </Fade>
+      <Collapse in={showInputField}>
+        <TextField
+          {...props}
+          className={classes.inputField}
+          onFocus={() => !showIcon && handleIconChange()}
```
Comment (gsulloa) [inline src/components/searchBar.tsx]: no usar funciones en linea

**Review by gsulloa — CHANGES_REQUESTED** [review]

**Review by gsulloa — APPROVED** [review]

**Review by gsulloa — CHANGES_REQUESTED** [review]
- Que en el primer elemento de la lista no aparezca la flecha
- Al presionar lupa, que haga focus
- Input sin lupa
- Acomodar input

File: src/components/searchBar.tsx [hunk 3]
```diff
@@ -13,7 +13,174 @@
...
+    },
+    [onSuggestedClick],
+  );
+
+  return (
+    <Box className={classes.box} display="block">
+      <List>
+        {searchResults.map((value) => (
+          <React.Fragment key={value}>
+            <ListItem key={value} href="#simple-list" button component="a">
```
Comment (gsulloa) [inline src/components/searchBar.tsx]: recibir callback dentro de una prop

**Review by coniurzua — APPROVED** [review]
Buen trabajo Luis, funciona bastante bien!

**Comment by mario-merino** [comment]
@gsulloa tu cierras este entonces?

**Pull request merged**